### ISSUE TITLE
[no-unused-vars] false positive on nested namespace declaration

### ISSUE THREAD

**Issue by dbeckwith** [typescript-eslint/typescript-eslint#2820]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
  }
}
```

```TS
// types.d.ts
export namespace a.b {
  export type X = string;
}
```

```TS
// main.ts
import { a } from './types';

const x: a.b.X = 'foobar';
```

**Expected Result**

`b` is clearly used, since removing it would cause a compile error, so no warning should appear

**Actual Result**

typescript-eslint reports a no-unused-vars warning on `b`, even though the namespace is used in the other file

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.8.1` |
| `@typescript-eslint/parser`        | `4.8.1` |
| `TypeScript`                       | `4.1.2` |
| `ESLint`                           | `7.13.0` |
| `node`                             | `15.1.0` |

**Comment by bradzacher** [comment]
duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/2573


please use the search. the previous issue title is even the same.
